feat: Update VAT Withholding fields and logic

SymonMuchemi opened the pull request:
This pull request focuses on improving the user experience and data accuracy of the VAT Withholding document within the CSF KE module, specifically concerning the **Withholder PIN** field.

The changes streamline data entry by automating the retrieval of the PIN when it is unavailable and provides clearer guidance to the user.

---

### Key Changes:

* **Automated PIN Retrieval:**
    * New backend logic ensures that if the `Withholder PIN` field is set to **"N/A"** (indicating the PIN is unavailable), the system will automatically attempt to fetch the correct Withholder PIN from the related **Sales Invoice**. This reduces manual effort and potential data entry errors.
    * A new `get_invoice_tax_id` method was introduced to reliably retrieve the Tax ID from the Sales Invoice.

* **User Guidance:**
    * The UI for the `withholder_pin` field has been updated to include a **placeholder** and a **description** instructing users to explicitly enter **"N/A"** when the PIN cannot be obtained.

* **Configuration:**
    * The document's `row_format` has been set to "Dynamic" to improve display flexibility.

These updates ensure more robust handling of missing Withholder PINs and provide a better, more guided experience for users completing the VAT Withholding document.

<img width="1580" height="270" alt="image" src="https://github.com/user-attachments/assets/8ec7240b-1752-46bb-b33e-6ab75b0bd9b9" />

emmanuel-mwendwa commented on csf_ke/csf_ke/doctype/vat_withholding/vat_withholding.json:
Why are you removing the reqd property?

emmanuel-mwendwa commented on csf_ke/csf_ke/doctype/vat_withholding/vat_withholding.js:
```diff
@@ -2,23 +2,43 @@
-            };
-        });
+  voucher_no(frm) {
+    if (frm.doc.voucher_no && frm.doc.customer === "Cash Customer") {
```
Don't hardcode the "Cash Customer" value, just look for the tax id from the transaction document, if there is no tax id in the customer doctype.

emmanuel-mwendwa commented on csf_ke/csf_ke/doctype/vat_withholding/vat_withholding.py:
```diff
@@ -115,3 +115,22 @@ def create_journal_entry(doc, method, *args, **kwargs):
+    invoice_data = frappe.db.get_value(
+        "Sales Invoice",
+        {"name": invoice_no, "company": company},
+        ["etr_invoice_number"],
```
You're provided with the ETR Number and ETR Serial No from KRA what you need to look for here is the Tax Id

SymonMuchemi committed:
feat(vat_withholding): enhance withholder pin handling and add tax ID retrieval

- Added a description and placeholder for the withholder pin field in the VAT Withholding JSON configuration.
- Updated the set_missing_values method to assign the withholder pin from the Sales Invoice tax ID if the pin is "N/A" and a voucher number is provided.
- Implemented a new method get_invoice_tax_id to retrieve the tax ID based on the voucher number, with error handling for missing tax IDs.
